### PR TITLE
Allow secrets in `AWS_Credential`

### LINKED ISSUE #8722: Ability to create an AWS_Credential with support for Enso Secrets
Want to be able to set the `access_key_id` or `secret_access_key` for an AWS Credential from an Enso Secret.

Currently only allows Text.

Should be able to then use within S3 reading as well.

Want to ensure the credentials provided is set up in std-aws polyglot not in Enso. Accept this is not 100% secure and a user could potential get the client and the keys out.

### PR DESCRIPTION
**Pull request by radeusgd**
### Pull Request Description

- Closes #8722 

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Review by GregoryTravis — APPROVED** [review]

File: std-bits/aws/src/main/java/org/enso/aws/ClientBuilder.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package org.enso.aws;
+
+import org.enso.base.enso_cloud.ExternalLibrarySecretHelper;
+import org.enso.base.enso_cloud.HideableValue;
+import software.amazon.awssdk.auth.credentials.*;
```
Comment (jdunkerley) [inline std-bits/aws/src/main/java/org/enso/aws/ClientBuilder.java]: can we avoid the * import please.

File: std-bits/base/src/main/java/org/enso/base/enso_cloud/ExternalLibrarySecretHelper.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+   *
+   * <p>This is a very rudimentary approach to the access control, and it is not very extensible, as
+   * it requires updating std-base whenever a new library that needs access to secrets is added.
+   * However, it seems like the best simple solution for now.
+   *
+   * <p>Later we may want to replace it with some other solution, e.g. a key that trusted libraries
+   * will use to 'sign' their class name, proving that they can be trusted, without the need to
+   * update std-base whenever a new library is added.
+   */
+  private static void checkAccess() throws EnsoSecretAccessDenied {
```
Comment (jdunkerley) [inline std-bits/base/src/main/java/org/enso/base/enso_cloud/ExternalLibrarySecretHelper.java:29]: do we know exactly how many steps up we need?
I think it is known to be up 2 frames.

**Review by jdunkerley — APPROVED** [review]

File: std-bits/base/src/main/java/org/enso/base/enso_cloud/ExternalLibrarySecretHelper.java (hunk 2 above)
Comment (radeusgd) [reply std-bits/base/src/main/java/org/enso/base/enso_cloud/ExternalLibrarySecretHelper.java:29]: Yes, I think you are right. I will try to change this into a constant.

File: std-bits/aws/src/main/java/org/enso/aws/ClientBuilder.java (hunk 1 above)
Comment (radeusgd) [reply std-bits/aws/src/main/java/org/enso/aws/ClientBuilder.java]: Sure, somehow it got through `javafmt`

File: lib/scala/json-rpc-server-test/src/main/scala/org/enso/jsonrpc/test/JsonRpcServerTestKit.scala [hunk 3]
```diff
@@ -153,7 +153,7 @@ abstract class JsonRpcServerTestKit
     }
 
     def expectSomeJson(
-      timeout: FiniteDuration = 5.seconds.dilated
```
Comment (hubertp) [inline lib/scala/json-rpc-server-test/src/main/scala/org/enso/jsonrpc/test/JsonRpcServerTestKit.scala:156]: OK, let's revert in #8806 fix

**Review by hubertp — APPROVED** [review]

**Pull request merged**